### PR TITLE
revert to xml string format method

### PR DESCRIPTION
**Pull request by Zurkdahool**
changed to XML string and fixed a minor bug

### CODE REVIEW

File: sources/indigo/export/materials/Specular.py [hunk 1]
```diff
@@ -62,8 +60,7 @@ def get_format(self):
             'name': [self.material_name],
             'backface_emit': [str(self.material_group.indigo_material_emission.backface_emit).lower()],
             element_name: {
-                #'internal_medium_name': [ medium_name ],
-                'internal_medium_uid': [ +medium_index + 10000 ]
+                'internal_medium_uid': [ medium_index + 10 ] # seems indigo medium uid starts at 10...
```
Comment (Ono-Sendai) [inline sources/indigo/export/materials/Specular.py]: Hi,
Indigo medium UIDs don't necessarily start at 10.  In fact there are no specific numbering rules for UIDs - they just have to be unique.

**Comment by Zurkdahool** [comment]
So i must doing something wrong while testing. I tested with very simple scene and edited by hand:

- uid 0:
Scene parsing error: Node with UID 0 was not a medium node.
- uid 1:
with a big scene it will work but duplicate entry in indigo gui. with simple scene error from above
- uid 2 - 7 it will work but Medium will shown twice in indigo gui (medium quick filter)
- uid >= 8 will work as expected

I can reproduce it. For example with uid 7:
twice
<code>
grep -i -C3 uid  debug.igs 

		<layer_index>0</layer_index>
	</layer_name>
	<medium>
		<uid>7</uid>
		<basic>
			<max_extinction_coeff>1.0</max_extinction_coeff>
			<ior>1.5</ior>
--
			<single_face>false</single_face>
			<arch_glass>false</arch_glass>
			<transparent>true</transparent>
			<internal_medium_uid>7</internal_medium_uid>
		</specular>
		<backface_emit>false</backface_emit>
		<name>Material</name>
</code>

**Comment by Ono-Sendai** [comment]
That's odd, can you email me the whole scene (as a .PIGS)?

**Comment by Ono-Sendai** [comment]
One thing to watch out for is to make sure the UID doesn't clash with other node UIDs.  If a node is not explicitly given a UID in the IGS, then a UID is automatically generated for it.  The auto-generated UID is just the next UID, e.g. max(all UIDs seen so far) + 1.

**Comment by Zurkdahool** [comment]
;-)  i found out after i prepared  the PIGS for you that uid's will be added automatically by indigo. In the original export from blendigo no uid for settings, metadata, camera and so on will be generated. After 'save as' UID 0-9 exists... I send you PIGS and orignal scene by mail... 
I think blendigo should generated uid for all nodes, or?! Why indigo does not respect the already take uid?

**Comment by Ono-Sendai** [comment]
Yeah, blendigo should probably generate UIDs for all nodes.

**Comment by Zurkdahool** [comment]
<code>
  <model2>
                <uid>10000000</uid>
</code>
where does this 10000000 come from?  :-)

**Comment by Zurkdahool** [comment]
Next version of blendigo will generate proper UIDs for nodes...

**Comment by Ono-Sendai** [comment]
Where does 10000000 come from?  Not sure, where did you find it?

**Comment by Zurkdahool** [comment]
generated after save the scene in indigo. Also this:
<code>
<material_uid>4294967295</material_uid>
</code>

also other duplicated uid was generated by indigo when saved as pigs (like camera)

**Comment by Zurkdahool** [comment]
i can reproduce it...

**Comment by Ono-Sendai** [comment]
If you have an Indigo scene with valid UIDs, and save it, and get invalid UIDs, that's an indigo bug.
If you find something like that, email it over.

**Comment by Zurkdahool** [comment]
ok ... i recommend to use the current workaround in blendigo (start uid  at 10). i  can remove uid and use name based (medium_name...) but this is not unique (duplicate names?!)

**Comment by Zurkdahool** [comment]
closing pull request until uid system is re-written

**Pull request closed without merging**